### PR TITLE
[1.1] Fix CI

### PR DESCRIPTION
**Pull request by kolyshkin**
CI fixes for `release-1.1` branch.

1. Seems that the F35 vagrant image was removed, so we need to switch to a supported Fedora version (37). This is roughly the same as backporting #3442 and #3670.

2. Backports of #3700, #3701 and #3526 (stolen from #3723).

Closes: #3723.

### CODE REVIEW

**Comment by kolyshkin** [comment]
ci / test failures are being addressed by https://github.com/opencontainers/runc/pull/3721

File: Vagrantfile.fedora [hunk 1]
```diff
@@ -23,7 +23,7 @@ Vagrant.configure("2") do |config|
       cat << EOF | dnf -y --exclude=kernel,kernel-core shell && break
 config install_weak_deps false
 update
-install iptables gcc make golang-go glibc-static libseccomp-devel bats jq git-core criu fuse-sshfs
+install iptables gcc golang-go make glibc-static libseccomp-devel bats jq git-core criu fuse-sshfs
```
Comment (thaJeztah) [inline Vagrantfile.fedora:26]: LOL, perhaps we should sort these alphabetically (assuming order doesn't matter here). At least that way we have something predictable.

**Review by thaJeztah — COMMENTED** [review]
> ci / test failures are being addressed by https://github.com/opencontainers/runc/pull/3723

Do we need to combine both PRs to get green CI?

File: Vagrantfile.fedora (hunk 1 above)
Comment (thaJeztah) [reply Vagrantfile.fedora:26]: Yeah, I gathered it was a backport, and mostly a nit / reminder to self on my side 😅

Comment (kolyshkin) [reply Vagrantfile.fedora]: OK, instead of backporting #3442 and maybe #3670, I just changed the version.

**Comment by kolyshkin** [comment]
> Do we need to combine both PRs to get green CI?

Yes, just did that here.

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
@AkihiroSuda ptal

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**